### ISSUE TITLE
Request: NPC Heatmaps.

### ISSUE THREAD

**Issue by IanWONeill** [runelite/runelite#1621]
It would be nice to have a way of looking up NPCs with their information, drop rates and other misc information as well as an optional heatmap to show where in the world the NPC has been spotted in terms of providing a heatmap.

There would need to be a collection of possible coordinates for NPCs spotted by players which would show up on the heat map. Collecting information on what NPCs have dropped after combat would also be good for droprates, provide acurate percentage aposed to 'common', 'rare' etc. which the typical fansites provide.

Just an idea which could be useful to player which no other Client has as far as I know.

**Comment by deathbeam** [comment]
This could be both useful and abusable, so I am not really sure if we want to implement this.

**Comment by devLotto** [comment]
The main issue I see is that people can submit fake data, so the data won't be 100% accurate?

**Comment by Unmoon** [comment]
Shouldn't be a problem for people to filter out the clearly bad data, especially if we release the data to public. Plus it will likely drown in the amount of real data. Wiki for example could really gain from something like this.

**Comment by IanWONeill** [comment]
I agree with Unmoon, even at 97-98% accuracy the mass data from real users would drown out the fake crap.

**Comment by TaxoRubio** [comment]
Is there a way of automating it? For example, when a foe dies its drop tile could be registered and when items were dropped there, it would update the db. That has a lot of error handling like players dropping fake items before the monster dies, but it could be worked around with a bit of wit.

**Comment by IanWONeill** [comment]
If made an opt-in Plugin then you could watch the players combat with NPCs, note the NPCs starting combat location, HP loss, Tile changes such as Ground item spawns, Animations used, Average Follow distance, Average Agg distance (note the players coordinate against the NPCs for when the fight is initiated.), Etc.

**Comment by IanWONeill** [comment]
https://rune-status.net/threads/everythingrs-heatmaps-realtime-osrs.2888/

Something similar with Players instead of NPCs. Though granted with alot less information.
